libs/langchain-community/src/vectorstores/libsql.ts

```diff
@@ -0,0 +1,221 @@
+
+  private readonly column: string;
+
+  /**
+   * Returns the type of vector store.
+   * @returns {string} The string "libsql".
+   */
+  _vectorstoreType(): string {
+    return "libsql";
+  }
+
+  /**
+   * Initializes a new instance of the LibSQLVectorStore.
+   * @param {EmbeddingsInterface} embeddings - The embeddings interface to use.
+   * @param {Client} db - The LibSQL client instance.
+   * @param {LibSQLVectorStoreArgs} options - Configuration options for the vector store.
+   */
+  constructor(
+    db: Client,
+    embeddings: EmbeddingsInterface,
+    options: LibSQLVectorStoreArgs = {
+      table: "vectors",
+      column: "embedding",
+    }
+  ) {
+    super(embeddings, options);
+
+    this.db = db;
+    this.table = options.table || "vectors";
+    this.column = options.column || "embedding";
+  }
+
+  /**
+   * Adds documents to the vector store.
+   * @param {Document[]} documents - The documents to add.
+   * @returns {Promise<string[]>} The IDs of the added documents.
+   */
+  async addDocuments(documents: Document[]): Promise<string[]> {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    const embeddings = await this.embeddings.embedDocuments(texts);
+
+    return this.addVectors(embeddings, documents);
+  }
+
+  /**
+   * Adds vectors to the vector store.
+   * @param {number[][]} vectors - The vectors to add.
+   * @param {Document[]} documents - The documents associated with the vectors.
+   * @returns {Promise<string[]>} The IDs of the added vectors.
+   */
+  async addVectors(
+    vectors: number[][],
+    documents: Document[]
+  ): Promise<string[]> {
+    const rows = vectors.map((embedding, idx) => ({
+      content: documents[idx].pageContent,
+      embedding: `[${embedding.join(",")}]`,
+      metadata: JSON.stringify(documents[idx].metadata),
+    }));
+
+    const batchSize = 100;
+    const ids: string[] = [];
+
+    for (let i = 0; i < rows.length; i += batchSize) {
+      const chunk = rows.slice(i, i + batchSize);
+      const insertQueries = chunk.map(
+        (row) =>
+          `INSERT INTO ${this.table} (content, metadata, ${this.column}) VALUES (${row.content}, ${row.metadata}, vector(${row.embedding})) RETURNING id`
+      );
+
+      const results = await this.db.batch(insertQueries);
+
+      for (const result of results) {
+        if (
+          result &&
+          result.rows &&
+          result.rows.length > 0 &&
+          result.rows[0].id != null
+        ) {
+          ids.push(result.rows[0].id.toString());
+        }
+      }
+    }
+
+    return ids;
+  }
+
+  /**
+   * Performs a similarity search using a vector query and returns documents with their scores.
+   * @param {number[]} query - The query vector.
+   * @param {number} k - The number of results to return.
+   * @returns {Promise<[Document, number][]>} An array of tuples containing the similar documents and their scores.
+   */
+  async similaritySearchVectorWithScore(
+    query: number[],
```

Comment:
Add a comment warning about potential SQL injection, or manually sanitize the input to make sure everything is a number.